### PR TITLE
Dagrun run taskinstance api

### LINKED ISSUE #23421: Add Run Task Instance endpoint to REST API
### Description

Replicate the run task instance endpoint [here](https://github.com/apache/airflow/blob/main/airflow/www/views.py#L1748) in the REST API.

POST `/api/v1/dags/{dag_id}/dagRuns/{dag_run_id}/taskInstances/{task_id}/run`

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by tirkarthi**
closes: https://github.com/apache/airflow/issues/23421
related: https://github.com/apache/airflow/issues/23421

supersedes PR : https://github.com/apache/airflow/pull/23530

### CODE REVIEW

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 1]
```diff
@@ -545,3 +550,86 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
...
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+        (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_TASK_INSTANCE),
+    ],
+)
+@provide_session
+def run_task_instance(
+    *, dag_id: str, dag_run_id: str, task_id: str, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Run a task instance."""
+    body = request.get_json()
```
Comment (bbovenzi) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py]:
```suggestion
    body = get_json_request_dict()
```

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 2]
```diff
@@ -545,3 +550,86 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
...
+    ignore_ti_state = data.get('ignore_ti_state')
+    map_index = data.get('map_index')
+
+    executor = ExecutorLoader.get_default_executor()
+    if not getattr(executor, "supports_ad_hoc_ti_run", False):
+        error_message = "Only works with the Celery, CeleryKubernetes or Kubernetes executors"
+        raise BadRequest(detail=error_message)
+
+    error_message = f"Dag ID {dag_id} not found"
+    dag = current_app.dag_bag.get_dag(dag_id)
```
Comment (bbovenzi) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py]:
```suggestion
    dag = get_airflow_app().dag_bag.get_dag(dag_id)
```

**Review by bbovenzi — CHANGES_REQUESTED** [review]
Found a few issues. Could you please test this endpoint on a real task instance?

Also, we should make the URL similar to a TI clear and mark state endpoints. Otherwise, I think developers will think the endpoint is missing in the docs.

**Comment by tirkarthi** [comment]
Thanks @bbovenzi , my branch was used outdated methods. I have rebased and incorporated your suggestions. I also ran test_task_instances.py to ensure there are no failures.

> Also, we should make the URL similar to a TI clear and mark state endpoints. Otherwise, I think developers will think the endpoint is missing in the docs.

Can you please elaborate on this? Does this change the endpoint parameters?

**Comment by bbovenzi** [comment]
> Can you please elaborate on this? Does this change the endpoint parameters?

Right now to clear or update a task instance, that lives here underneath the DAG endpoints. Since they allow for updating/clearing multiple task instances at once:
Screen Shot 2022-09-22 at 11 58 43 AM

But right now, to run a task instance the url is different and it only takes a single task instance. @ephraimbuddy what do you think? Should we keep this PR as-is or should it be changed to clear multiple task instances at once like run/update are?

**Pull request closed without merging**